### PR TITLE
[fix](httpapi) restore compaction/run_status api can show be's overall compaction status and refactor code

### PR DESCRIPTION
**Pull request by Yulei-Yang**
## Proposed changes

Issue: http api compaction/run_status without parameter does not work now, like below：
```sql
curl -X GET http://localhost:8040/api/compaction/run_status
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0{
    "status": "INTERNAL_ERROR",
    "msg": "tablet id and table id can not be empty at the same time!"
```
while in https://doris.apache.org/zh-CN/docs/admin-manual/be/compaction-status
企业微信截图_70dcf0b0-06d7-419d-a498-0fef9aa7a4cf

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run buildall

File: be/src/http/action/compaction_action.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ CompactionAction::CompactionAction(CompactionActionType ctype, ExecEnv* exec_env
                                    TPrivilegeType::type ptype)
         : HttpHandlerWithAuth(exec_env, hier, ptype), _engine(engine), _type(ctype) {}
 
-/// check param and fetch tablet_id from req
+/// check param and fetch tablet_id & table_id from req
```
Comment (xy720) [inline be/src/http/action/compaction_action.cpp:64]: You should modify the comment in cloud compaction action too

File: be/src/http/action/compaction_action.cpp [hunk 2]
```diff
@@ -178,10 +195,8 @@ Status CompactionAction::_handle_run_compaction(HttpRequest* req, std::string* j
 
 Status CompactionAction::_handle_run_status_compaction(HttpRequest* req, std::string* json_result) {
     uint64_t tablet_id = 0;
-    uint64_t table_id = 0;
-
-    // check req_tablet_id is not empty
-    RETURN_NOT_OK_STATUS_WITH_WARN(_check_param(req, &tablet_id, &table_id), "check param failed");
+    RETURN_NOT_OK_STATUS_WITH_WARN(_check_param(req, &tablet_id, TABLET_ID_KEY),
```
Comment (xy720) [inline be/src/http/action/compaction_action.cpp:198]: please add some comments here.

**Review by xy720 — COMMENTED** [review]
Please also update the document.

File: be/src/http/action/compaction_action.cpp (hunk 1 above)
Comment (Yulei-Yang) [reply be/src/http/action/compaction_action.cpp:64]: done

File: be/src/http/action/compaction_action.cpp (hunk 2 above)
Comment (Yulei-Yang) [reply be/src/http/action/compaction_action.cpp:200]: done

**Comment by Yulei-Yang** [comment]
run buildall

**Review by xy720 — DISMISSED** [review]
LGTM

**Review by lide-reed — DISMISSED** [review]
LGTM

**Review by dataroaring — CHANGES_REQUESTED** [review]
Please add a regression tests and the document.

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
> Please add a regression tests and the document.

Regression test added. And document is added in this pr's description.Cloud you pls review again?

**Review by cambyzju — APPROVED** [review]
LGTM

**Review by lide-reed — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**